server/controllers/users.js

```diff
@@ -1,12 +1,32 @@
-// import will go here
-
-// the class for the user
+import crypto from 'crypto';
+import userModel from '../models/users';
+import auth from '../middleware/Auth';
 
 const user = {
-    // methodes will go here and you need to follow the same method
-    async createUser () {
-        console.log('the creation of the user')
+  async createUser(req, res) {
+    const { email, password } = req.body;
+    const salt = crypto.randomBytes(16).toString('hex');
+    const hash = crypto.pbkdf2Sync(password, salt, 1000, 64, 'sha512').toString('hex');
+    const response = await userModel.createUser({ email, salt, hash });
+
+    if (response.rowCount === 1) {
+      const userData = response.rows[0];
+      const { userId } = userData;
+      const token = auth.generateToken({ email, userId });
+
+      res.status(201).json({
+        status: 201,
+        token,
+        user: userData,
+      });
+      return;
     }
-}
+    const message = (response.code === '23505') ? `email ${req.body.email} already exists` : response.detail;
```

Comment:
For me, I have avoided using ternary operators because of sometimes the status code based on the returned error message.  (For this case u can check if they are different or the same)